### PR TITLE
Bug 1992596: e2e/cli: Finish basicresources.sh migration

### PR DESCRIPTION
**Pull request by deejross**
This moves the last of the legacy basicresources.sh tests to Go tests, including expose and template processing.

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: test/extended/testdata/bindata.go [hunk 1]
```diff
@@ -28852,153 +28851,6 @@ func testExtendedTestdataCmdTestCmdAuthenticationSh() (*asset, error) {
...
-# User-Agent information from our clients regardless of which resources they're trying to access
-#os::build::version::get_vars
-#os_git_regex="$( escape_regex "${OS_GIT_VERSION%%-*}" )"
-#kube_git_regex="$( escape_regex "${KUBE_GIT_VERSION%%-*}" )"
-#etcd_version="$(echo "${ETCD_GIT_VERSION}" | sed -E "s/\-.*//g" | sed -E "s/v//")"
-#etcd_git_regex="$( escape_regex "${etcd_version%%-*}" )"
-#os::cmd::expect_success_and_text 'oc version' "Client Version: .*GitVersion:\"${os_git_regex}"
-#os::cmd::expect_success_and_text 'oc version' "Server Version: .*GitVersion:\"${kube_git_regex}"
-#os::cmd::expect_success_and_text "curl -k '${API_SCHEME}://${API_HOST}:${API_PORT}/version'" "${kube_git_regex}"
-#os::cmd::expect_success_and_text "curl -k '${API_SCHEME}://${API_HOST}:${API_PORT}/version'" "${OS_GIT_COMMIT}"
```
[the remark this answers is not in the document]
Comment (deejross) [reply test/extended/testdata/bindata.go:28891]: Sure, I've been trying to add more tests on top of the ones being migrated. I'll make a note of that and include in a future PR.

**Comment by deads2k** [comment]
this broke payload promotion.  see https://github.com/openshift/origin/pull/26597 for details

**Pull request merged**